src/Core/PotreeNode.js

```diff
@@ -80,28 +80,26 @@ class PointCloudNode {
         throw new Error(`Cannot find node with name '${name}'`);
     }
 
-    octreeUrl() {
-        return `${this.baseurl}/r${this.name}.${this.layer.source.extensionOctree}`;
-    }
-
-    nodeUrl() {
-        return `${this.baseurl}/r${this.name}.${this.layer.source.extension}?isleaf=${this.isLeaf ? 1 : 0}`;
-    }
-
     get octreeIsLoaded() {
         return !(this.childrenBitField && this.children.length === 0);
     }
 
-    get isLeaf() {
-        return this.childrenBitField == 0;
+    get url() {
+        return `${this.baseurl}/r${this.name}.${this.layer.source.extension}`;
     }
 
-    loadNode() {
-        return this.layer.source.fetcher(this.nodeUrl(), this.layer.source.networkOptions).then(this.layer.source.parse);
+    load() {
+        // Query octree/HRC if we don't have children potreeNode yet.
+        if (!this.octreeIsLoaded) {
+            this.loadOctree();
```

Comment:
How do you handle the `Promise` part ?

Reply:
the promise part had never handling in `Provider`. I don't understand why.